pkg/sockops/sockops.go

```diff
@@ -0,0 +1,491 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sockops
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/datapath/loader"
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/mountinfo"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// Path to where cgroup is mounted
+	cgroupRoot = defaults.DefaultCgroupRoot
+
+	// Set to true on first get request to detect misorder
+	cgrpLockedDown = false
+
+	// Only mount a single instance
+	cgrpMountOnce sync.Once
+
+	// Default prefix for map objects
+	mapPrefix = defaults.DefaultMapPrefix
+
+	// Set to true after sockopsRoot beomes in-use
+	sockopsLockedDown = false
```

Comment:
Do you want to wrap this variables inside a sync.Once?

Reply:
I'm going to just drop the LockedDown variables they only matter if I add back the runtime support for enabling/disabling sockmap I did earlier. I've dropped support for this now aiming for a smaller patchset in the original series. Also not clear if even would want runtime enable/disable support but that can be debated later.